core/crypto/abac/attributes_test.go

```diff
@@ -0,0 +1,229 @@
+/*
+Copyright IBM Corp. 2016 All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+		 http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package abac
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/hyperledger/fabric/core/crypto/primitives"
+)
+
+func TestMain(m *testing.M) {
+	if err := primitives.InitSecurityLevel("SHA3", 256); err != nil {
+		fmt.Printf("Failed setting security level: %v", err)
+	}
+
+	ret := m.Run()
+	os.Exit(ret)
+}
+
+func TestEncryptDecryptAttributeValuePK0(t *testing.T) {
+	expected := "ACompany"
+
+	preK0 := []byte{
+		91, 206, 163, 104, 247, 74, 149, 209, 91, 137, 215, 236,
+		84, 135, 9, 70, 160, 138, 89, 163, 240, 223, 83, 164, 58,
+		208, 199, 23, 221, 123, 53, 220, 15, 41, 28, 111, 166,
+		28, 29, 187, 97, 229, 117, 117, 49, 192, 134, 31, 151}
+
+	encryptedAttribute, err := EncryptAttributeValuePK0(preK0, "company", []byte(expected))
+	if err != nil {
+		t.Error(err)
+	}
+
+	attributeKey := getAttributeKey(preK0, "company")
+
+	attribute, err := DecryptAttributeValue(attributeKey, encryptedAttribute)
+	if err != nil {
+		t.Error(err)
+	}
+
+	if string(attribute) != expected {
+		t.Errorf("Failed decrypting attribute. Expected: %v, Actual: %v", expected, attribute)
+	}
+}
+
+func TestGetKAndValueForAttribute(t *testing.T) {
+	expected := "Software Engineer"
+
+	tcert, prek0, err := loadTCertAndPreK0()
+	if err != nil {
+		t.Error(err)
+	}
+
+	_, attribute, err := getKAndValueForAttribute("position", prek0, tcert)
```

Comment:
why are we (only) testing private functions? Seems to me that we should be testing both, but most importantly the public functions. Also, we only have positive tests here. Should have tests that ensure the correct processing of calls that pass incorrect argument values.